test/functional/lua/vim_spec.lua

```diff
@@ -175,13 +182,6 @@ describe('lua stdlib', function()
         )
       end)
 
-      it('plugin=nil, to be deleted in the next major version (1.0)', function()
-        eq(
-          [[foo.baz() is deprecated. Run ":checkhealth vim.deprecated" for more information]],
-          exec_lua [[ return vim.deprecate('foo.baz()', nil, '1.0') ]]
```

Comment:
Not sure if this was intentional, but doesn't really make sense. Why would we soft-deprecate `vim.deprecate(...,'0.12')` but hard-deprecate `vim.deprecate(...,'1.0')` ?

Reply:
[Intentional, but still sus IMO](https://github.com/neovim/neovim/pull/27027#discussion_r1455891052). I think I'm with you that 1.0 should be soft-deprecated.